LinSerial.cpp

```diff
@@ -96,438 +81,305 @@ Serial::Serial(const char* Port, const SerParam& SP, const SerTimeOut& ST){
-		break;
-	case SerParam::Baud110:
-		std::cout << "B110" << std::endl;
-		break;
-	case SerParam::Baud134:
-		std::cout << "B134" << std::endl;
-		break;
-	case SerParam::Baud150:
-		std::cout << "B150" << std::endl;
-		break;
-	case SerParam::Baud200:
-		std::cout << "B200" << std::endl;
-		break;
-	case SerParam::Baud300:
-		std::cout << "B300" << std::endl;
-		break;
-	case SerParam::Baud600:
-		std::cout << "B600" << std::endl;
-		break;
-	case SerParam::Baud1200:
-		std::cout << "B1200" << std::endl;
-		break;
-	case SerParam::Baud1800:
-		std::cout << "B1800" << std::endl;
-		break;
-	case SerParam::Baud2400:
-		std::cout << "B2400" << std::endl;
-		break;
-	case SerParam::Baud4800:
-		std::cout << "B4800" << std::endl;
-		break;
-	case SerParam::Baud9600:
-		std::cout << "B9600" << std::endl;
-		break;
-	case SerParam::Baud19200:
-		std::cout << "B19200" << std::endl;
-		break;
-	case SerParam::Baud38400:
-		std::cout << "B38400" << std::endl;
-		break;
-	case SerParam::Baud57600:
-		std::cout << "B57600" << std::endl;
-		break;
-	case SerParam::Baud115200:
-		std::cout << "B115200" << std::endl;
-		break;
-	case SerParam::Baud230400:
-		std::cout << "B230400" << std::endl;
-		break;
-	case SerParam::Baud460800:
-		std::cout << "B460800" << std::endl;
-		break;
-	default:
-		break;
-	}
-	#endif
+	linSerLogInfo("Baudrate set to: %d\n", serParam::baudrateString[SP.rate]);
 
 	cfsetspeed(&tty, (speed_t)SP.rate);
 
 	// Save tty settings, also checking for error
 	if (tcsetattr(hSerial, TCSANOW, &tty) != 0) {
-		#if LINSERDEBUG >= 1
-		std::cout << "[LinSer] Error " << std::to_string(errno) << "from tcsetattr(): " << std::strerror(errno) << std::endl;
-		#endif
-		throw SerialException(
+		linSerLogError("%s from tcsetattr(): %s\n", std::to_string(errno).c_str(), std::strerror(errno));
+		throw serialException(
 			"Error " + std::to_string(errno) +
 			"from tcsetattr(): " + std::strerror(errno),
-			SerialException::tcsetattrError);
+			serialException::tcsetattrError);
 	}
 
-	usleep(10000); //required to make flush work, for some reason
+	usleep(10000); // Required to make flush work, for some reason
   	tcflush(hSerial,TCIOFLUSH);
 
-	// start threads
-	IncomingBuffer.ThreadFunc = std::thread(ReadThreadFunc, std::ref(IncomingBuffer), std::ref(hSerial), std::ref(SerialHandleMutex));
+	// Start threads
+	incomingThread = std::thread(_readThreadFunc, std::ref(*this));
 }
 
 
 
-Serial::~Serial(){
-	#if LINSERDEBUG >= 3
-	std::cout << "[LinSer] Closing Serial thread" << std::endl;
-	#endif
-	IncomingBuffer.StopThread = true;
-	IncomingBuffer.ThreadFunc.join();
+serial::~serial(){
+	linSerLogInfo("Closing serial thread...\n");
+	stopThread = true;
+	incomingThread.join();
 	
-	SerialHandleMutex.lock();
+	serialHandleMutex.lock();
 	close(hSerial);
-	SerialHandleMutex.unlock();
+	serialHandleMutex.unlock();
 	hSerial = 0;
 
-	#if LINSERDEBUG >= 3
-	std::cout << "[LinSer] Serial thread closed" << std::endl;
+	#if LINSER_LOGLEVEL >= 3
```

Comment:
most likely a problem from the current development branch but this pre processor macro should not be used, also wrong define name if intended.